### ISSUE TITLE
Map from react-native-maps doesn't load in Android if expo-location included in development build

### ISSUE THREAD

**Issue by charlesfsl** [expo/expo#21103]
### Summary

The map from `react-native-maps` does not load in Android If the `expo-location` package is included in the `package.json` and the app is run in a dev client. If `loadingEnable={true}` on the `MapView` component, you see a spinner indefinitely. This is not an issue in iOS. The following combinations work, or not:

`react-native-maps` + `expo-location` → Works!
`react-native-maps` + dev client → Works!
`react-native-maps` + `expo-location` + dev client → Does not work!

There are no errors messages, just a silent fail. I did add the development build's SHA-1 fingerprint to the Google Maps API key's allowed list for the package name. 

While this is part of a larger project, I have reproduced the issue in a simple app with only these packages. 

### Managed or bare workflow?

managed

### What platform(s) does this occur on?

Android

### Package versions

```json
{
  "dependencies": {
    "expo": "~47.0.12",
    "expo-dev-client": "~2.0.1",
    "expo-location": "~15.0.1",
    "expo-status-bar": "~1.4.2",
    "react": "18.1.0",
    "react-native": "0.70.5",
    "react-native-maps": "1.3.2"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9"
  },
}
```

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.15 NixOS 22.11 (Raccoon) 22.11 (Raccoon)
      Shell: 3.5.1 - /run/current-system/sw/bin/fish
    Binaries:
      Node: 18.12.1 - /run/current-system/sw/bin/node
      Yarn: 1.22.19 - /run/current-system/sw/bin/yarn
      npm: 8.19.2 - /run/current-system/sw/bin/npm
    npmPackages:
      expo: ~47.0.12 => 47.0.13
      react: 18.1.0 => 18.1.0
      react-native: 0.70.5 => 0.70.5
    npmGlobalPackages:
      eas-cli: 3.5.1
      expo-cli: 6.0.8
    Expo Workflow: managed
```

### Reproducible demo

A gist of my `App.js`: https://gist.github.com/charlesfsl/7e1febe2024098057ae1b3b55eb5368d

1. Create a new expo app
2. replace the App.js with the above gist
3. install the packages I listed above
4. run `npx eas-cli build --profile development` following the dev client documentation
5. setup an API key with Google Maps that's restricted to Android apps with you new app's package name, and be sure to add the SHA-1 fingerprint from Expo for this package
6. install the development client on your Android device and open it up
7. View a white screen with an indefinite loading spinner

**Comment by Jamestavs** [comment]
Any luck resolving this? Having the same issue...

**Comment by charlesfsl** [comment]
@Jamestavs I did. Kim in the Expo Developers Discord was right to point to the SHA-1 fingerprint used to authorize use of the maps API, though I did have that setup correctly. The fix was to remove the package name and corresponding SHA-1 fingerprint from the Google API key config and then add it back. Exact same package name and SHA-1 before and after. 

One thing to note that wasn't clear to me at first, though it's in the docs, is that you obtain the SHA-1 fingerprint from different places depending on if you're building the app to put in an app store or if you're running a local development instance. Make sure to not miss the two tabs in Expo's MapView docs' section about deploying with Google Maps: https://docs.expo.dev/versions/v47.0.0/sdk/map-view/#copy-your-apps-sha-1-certificate-fingerprint

**Comment by Said20hr** [comment]
@charlesfsl , thank you , in my case the issue was in the SH1, I did not copied correctly

**Comment by mschipperheyn** [comment]
So, for those finding this issue. Some things have to be in place:
- Create a maps Api key in Google Developer console
- Your Google Developer console maps Api key has to be placed in `app.config.ts` => `android: { config: { googleMaps: { apiKey: 'XYZ' } } }
- You need to download your build credentials from Google Play Console
- Your build credentials need to be uploaded to expo using `eas credentials`
- The SHA-1 from your build credentials needs to be placed in Android restrictions in the Google developer console maps key configuration using `[package name]: [SHA-1]`, e.g. com.mydomain.appName: AB:B0:C1...
- Copy these settings over to your android build using `eas prebuild`
- Build your app using `eas build -p android --profile production` (profile depending on deploy target)

That should be it.

**Comment by karimdakroury** [comment]
Hi, my MapView doesn't load. I can only see an infinite loading spinner on the screen. I have SHA-1 set up and it is working on expo go but not on my Android build.

I have a question, i have 2 google accounts. 1 for the google play store and the other account with the API credentials. I am currently using the SHA-1 from the account with API credentials and trying to release the build from the account with the google play store. 

Do you think because the SHA-1 is not from the google play store account thats why its not working?

**Comment by vaishnavipy** [comment]
@karimdakroury  I have the same exact issue. Infinitely loading spinner.. any solution to this issue? Please let me know, I've been stuck with this issue for way too long.

**Comment by karimdakroury** [comment]
> @karimdakroury I have the same exact issue. Infinitely loading spinner.. any solution to this issue? Please let me know, I've been stuck with this issue for way too long.

@vaishnavipy on Expo Go everything is working on both IOS and Android. For my production build, it is working on IOS but not Android. My Location.getCurrentPositionAsync({}) does not get called and hence everything stops working.

**Comment by karimdakroury** [comment]
> @Jamestavs I did. Kim in the Expo Developers Discord was right to point to the SHA-1 fingerprint used to authorize use of the maps API, though I did have that setup correctly. The fix was to remove the package name and corresponding SHA-1 fingerprint from the Google API key config and then add it back. Exact same package name and SHA-1 before and after.
> 
> One thing to note that wasn't clear to me at first, though it's in the docs, is that you obtain the SHA-1 fingerprint from different places depending on if you're building the app to put in an app store or if you're running a local development instance. Make sure to not miss the two tabs in Expo's MapView docs' section about deploying with Google Maps: https://docs.expo.dev/versions/v47.0.0/sdk/map-view/#copy-your-apps-sha-1-certificate-fingerprint

@charlesfsl in my play store app signing their are 2 SHA-1 App Signing and uplaod key which should i use?

**Comment by jlsnjnr** [comment]
Guys, besides the SHA-1 settings, what helped me fix the behavior was changing the 'initialRegion' to just 'region'
` <MapView
          loadingEnabled
          provider={PROVIDER_GOOGLE}
          ref={mapRef}
          className="w-full h-screen flex-1"
          region={
            {
              latitude: -23.179363,
              longitude: -45.885344,
              latitudeDelta: 0.01,
              longitudeDelta: 0.01,
            }
          }
/>`

**Comment by FranciscoOssian** [comment]
Sorry if I misunderstood. But from what I gathered, you encountered the error after leaving the managed workflow provided by Expo, meaning you no longer use Expo Go. If that's the case, it seems normal. I had this problem recently when, without much thought, I installed a Firebase library and did a prebuild for that. If this is indeed the case, it might be because only Expo Go holds the development key used for you to visualize the map without needing to use your own key. I imagine that's the issue.
